src/Transpiler.ts

```diff
@@ -1279,6 +1277,39 @@ export abstract class LuaTranspiler {
         }
     }
 
+    public transpileElementCall(node: ts.CallExpression): string {
+        if (!ts.isElementAccessExpression(node.expression)) {
+            throw TSTLErrors.InvalidPropertyCall(node);
+        }
+
+        const signature = this.checker.getResolvedSignature(node);
+        let parameters = this.transpileArguments(node.arguments, signature);
+
+        const signatureDeclaration = signature.getDeclaration();
+        if (!signatureDeclaration
+            || tsHelper.getDeclarationContextType(signatureDeclaration, this.checker) !== ContextType.Void) {
+            // Pass left-side as context
+            if (node.arguments.length > 0) {
```

Comment:
This looks a little weird but I do understand it prevents some duplication. Maybe if you just improve the comment a little bit on the logic for this `if` statement, that would already help.